wave_vision/include/wave/vision/matcher/descriptor_matcher.hpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file
+ * Base class from which descriptor matchers can be derived.
+ * @ingroup vision
+ */
+#ifndef WAVE_VISION_DESCRIPTOR_MATCHER_HPP
+#define WAVE_VISION_DESCRIPTOR_MATCHER_HPP
+
+/** C++ Headers */
+#include <exception>
+
+/** Third Party Headers */
+#include <opencv2/opencv.hpp>
+
+/** Libwave Headers */
+#include "wave/utils/utils.hpp"
+
+namespace wave {
+/** @addtogroup vision
+ *  @{ */
+
+/** Representation of a generic descriptor matcher.
+ *
+ *  Internally, this class, and all derived classes are wrapping various
+ *  descriptor matchers implemented in OpenCV. Further reference on descriptor
+ *  matchers can be found [here][opencv_descriptor_matchers].
+ *
+ *  [opencv_descriptor_matchers]:
+ *  http://docs.opencv.org/trunk/db/d39/classcv_1_1DescriptorMatcher.html
+ */
+
+class DescriptorMatcher {
+ protected:
+    /** Destructor */
+    ~DescriptorMatcher() = default;
```

Comment:
Using this as reference: https://stackoverflow.com/questions/13576055/how-is-default-different-from-for-default-constructor-and-destructor

It apparently helps differentiate between a _trivial_ and _non-trivial_ class. If I specified the destructor as `~DescriptorMatcher() {}`, it is no longer trivial, due to a user-specified destructor. By stating `~DescriptorMatcher() = default;`, I'm letting the compiler automatically generate the default destructor.

Now perhaps @leokoppel has more to add on my baseline understanding. But as far as I can tell that the main difference has to do with class triviality. My change from `~DescriptorMatcher() {}` to `~DescriptorMatcher() = default;` was a suggestion made in a previous PR.